packages/block-library/src/block/index.php

```diff
@@ -46,8 +46,26 @@ function render_block_core_block( $attributes ) {
 	$content = $wp_embed->run_shortcode( $reusable_block->post_content );
 	$content = $wp_embed->autoembed( $content );
 
+	/**
+	 * We set the `overrides` context through the `render_block_context`
+	 * filter so that it is available when a pattern's inner blocks are
+	 * rendering via do_blocks given it only receives the inner content.
+	 */
+	if ( isset( $attributes['overrides'] ) ) {
+		$filter_block_context = static function ( $context ) use ( $attributes ) {
+			$context['overrides'] = $attributes['overrides'];
+			return $context;
+		};
+		add_filter( 'render_block_context', $filter_block_context, 1 );
+	}
+
 	$content = do_blocks( $content );
 	unset( $seen_refs[ $attributes['ref'] ] );
+
+	if ( isset( $attributes['overrides'] ) ) {
+		remove_filter( 'render_block_context', $filter_block_context, 1 );
+	}
```

Comment:
Should we check the experiment flag before adding the block context?

I guess right now it depends on the existence of `overrides`, is there any chance that will still exist once the experiment is switched off?